### PR TITLE
feat: combine send and send batch functions

### PR DESCRIPTION
**Pull request by HomelessDinosaur**

### CODE REVIEW

**Review by jyecusch — COMMENTED** [review]
Hmm, I can see the appeal of this. However, I'm not sure I love the idea of getting an array of FailedTasks back as a response to a request to send a single item to the queue. I'd like to hear other people's thoughts on that.

**Comment by tjholm** [comment]
> Hmm, I can see the appeal of this. However, I'm not sure I love the idea of getting an array of FailedTasks back as a response to a request to send a single item to the queue. I'd like to hear other people's thoughts on that.

After looking at this and the alternatives we should implement this as a method overload (multiple signatures with a single implementation). Then the typescript compiler can infer the return type from the style of call. Before only a `void | array` union type was available which meant that this method could only safely be used by interrogating the output type for each invocation.

```typescript
async function send(tasks: NitricTask): Promise<void>;
async function send(tasks: NitricTask[]): Promise<FailedTask[]>;
// NOTE: Not sure if we can union the internal promise without breaking the above typing.
async function send(tasks: NitricTask | NitricTask[]): Promise<void> | Promise<FailedTask[]> {
   ...implementation
}
```
see: https://www.typescriptlang.org/docs/handbook/functions.html#overloads

File: src/api/queues/v0/queues.ts [hunk 1]
```diff
@@ -104,80 +104,35 @@ export class Queue {
...
-   * }]);
-   *
-   * // do something with failedTasks
-   * // console.log(failedTasks);
-   * ```
    */
-  private sendBatch = async (tasks: Task[]): Promise<FailedMessage[]> => {
+   async send(tasks: Task): Promise<FailedMessage>;
+   async send(tasks: Task[]): Promise<FailedMessage[]>;
+   async send(tasks: Task | Task[]): Promise<FailedMessage | FailedMessage[]> {
```
Comment (jyecusch) [inline src/api/queues/v0/queues.ts:110]: Just chatting to Tim, we were thinking the single task implementation should probably have a `void` response (and throw an exception if the send fails). That will provide more consistency with other methods like publishing events.

**Review by jyecusch — CHANGES_REQUESTED** [review]

File: src/api/queues/v0/queues.ts (hunk 1 above)
Comment (tjholm) [reply src/api/queues/v0/queues.ts:110]: just did some quick reading apparently `never` might be more conventional in the case where we throw an error:
```typescript
async send(tasks: Task): Promise<never>;
```

Comment (tjholm) [reply src/api/queues/v0/queues.ts]: makes sense, go with `void`.

**Review by jyecusch — APPROVED** [review]

**Pull request merged**